src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -121,58 +154,99 @@ def bulk_record(
-                organization_id, string = write_pair
-                new_records.append(
-                    self._table(use_case_id)(organization_id=int(organization_id), string=string)
-                )
+            if use_case_path_key is UseCaseKey.PERFORMANCE:
+                new_records = [
+                    self._get_table_from_use_case_ids(strings.keys())(
+                        organization_id=int(organization_id),
+                        string=string,
+                        use_case_id=use_case_id.value,
+                    )
+                    for organization_id, string in filtered_db_write_keys.as_tuples()
+                ]
+            else:
+                new_records = [
+                    self._get_table_from_use_case_ids(strings.keys())(
+                        organization_id=int(organization_id),
+                        string=string,
+                    )
+                    for organization_id, string in filtered_db_write_keys.as_tuples()
+                ]
 
             with metrics.timer("sentry_metrics.indexer.pg_bulk_create"):
-                self._bulk_create_with_retry(self._table(use_case_id), new_records)
+                self._bulk_create_with_retry(
+                    self._get_table_from_use_case_ids(strings.keys()), new_records
+                )
 
-        db_write_key_results = KeyResults()
-        db_write_key_results.add_key_results(
+        db_write_key_results = UseCaseKeyResults()
+        db_write_key_results.add_use_case_key_results(
             [
-                KeyResult(org_id=db_obj.organization_id, string=db_obj.string, id=db_obj.id)
-                for db_obj in self._get_db_records(use_case_id, filtered_db_write_keys)
+                UseCaseKeyResult(
+                    use_case_id,
+                    org_id=db_obj.organization_id,
+                    string=db_obj.string,
+                    id=db_obj.id,
+                )
+                for db_obj in self._get_db_records(
+                    UseCaseKeyCollection({use_case_id: filtered_db_write_keys})
+                )
             ],
             fetch_type=FetchType.FIRST_SEEN,
         )
 
         return db_read_key_results.merge(db_write_key_results).merge(rate_limited_key_results)
 
-    def record(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
-        """Store a string and return the integer ID generated for it"""
-        result = self.bulk_record(use_case_id=use_case_id, org_strings={org_id: {string}})
-        return result[org_id][string]
+    def _uca_record(self, use_case_id: UseCaseID, org_id: int, string: str) -> Optional[int]:
+        result = self._uca_bulk_record(strings={use_case_id: {org_id: {string}}})
+        return result[use_case_id][org_id][string]
 
     def resolve(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
         """Lookup the integer ID for a string.
```

Comment:
just double checking -- the old PR (with postgres changes) has 

```table = self._get_table_from_metric_path_key(use_case_id)```

this is the same as what you have in this PR right?